### PR TITLE
Make Slack integration work with agentless alerts

### PR DESCRIPTION
**Pull request by dmitryax**
## Description

Agentless configuration doesn't work with Slack integration. Slack webhooks not triggered and you can see error in logs.   

## Logs/Alerts example

Error logs:
```
2019/06/18 20:14:19 ossec-integratord: ERROR: Unable to run integration for slack -> /var/ossec/integrations/slack
2019/06/18 20:14:19 ossec-integratord: ERROR: While running slack -> /var/ossec/integrations/slack. Output: KeyError: 'agent'
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] MAC OS X
- [x] Source installation
- [x] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- Memory tests
  - [ ] Valgrind report for affected components
  - [ ] CPU impact
  - [ ] RAM usage impact
- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

### CODE REVIEW

**Comment by hmbg** [comment]
Please let me know how can I improve the PR if it's not descriptive enough. Also didn't find a way to run tests.

**Comment by Zenidd** [comment]
Hi again @hmbg,

I've opened an issue linked to this PR with the steps to reproduce the error. Let's review it and if possible add it to the Wazuh project.

Thank you very much for your contribution, it's great!
<br></br>
Greetings,

Juan Pablo Sáez

File: integrations/slack [hunk 1]
```diff
@@ -76,21 +76,26 @@ def generate_msg(alert):
...
-    agent = { "title":"Agent", "value":"({0}) - {1}".format(alert['agent']['id'],alert['agent']['name']) }
-    location = { "title":"Location", "value":alert['location'] }
-    rule = { "title":"Rule ID", "value":"{0} _(Level {1})_".format(alert['rule']['id'],level) }
-    msg['fields'] = [ agent, location, rule ]
+    
+    msg['fields'] = []
+    if 'agent' in alert:
+        msg['fields'].append({ "title": "Agent", "value": "({0}) - {1}".format(alert['agent']['id'],alert['agent']['name']) })
+    if 'agentless' in alert:
+        msg['fields'].append({ "title": "Host", "value": alert['agentless']['host'] })
```
Comment (Zenidd) [inline integrations/slack]: - could you extend the "title" from `"Host"` to `"Agentless Host"` to make it more readable?

- Please, could you validate the code with http://pep8online.com and make the style changes indicated?

Kind regards,

Juan Pablo Sáez

Comment (hmbg) [reply integrations/slack]: @Zenidd thanks for response. The comments are addressed

**Comment by hmbg** [comment]
> Please check these two little details. Your PR has been queued to be merged on Wazuh 3.11
> Also, change the target branch of your PR from `wazuh:master` to `wazuh:3.11`.

I changed the target branch, resolved conflicts, but one file wazuh-installer.wxs still marked as changed for some reason, even if looks exactly the same

**Comment by Zenidd** [comment]
Hi @hmbg,

I just checked it out and it's great! The python validator has no complaints and  you've left the code cleaner than you found it.

That's how an agentless alert looks on Slack app after your fixes:

agentlesshost


We're going to merge it as soon as possible. Thank you very much for your contribution to the Wazuh project.
<br></br>
Best regards,

Juan Pablo Sáez

**Comment by hmbg** [comment]
@Zenidd Thank you for support. Happy to help.

P.S. Pushed a small fix for color assignment.

**Comment by dmitryax** [comment]
Hi @chemamartinez , I rebased and submitted another PR https://github.com/wazuh/wazuh/pull/3971 
Is it Ok?

**Comment by JuantAldea** [comment]
> > Please check these two little details. Your PR has been queued to be merged on Wazuh 3.11
> > Also, change the target branch of your PR from `wazuh:master` to `wazuh:3.11`.
> 
> I changed the target branch, resolved conflicts, but one file wazuh-installer.wxs still marked as changed for some reason, even if looks exactly the same

This is because you saved the file converting the newlines.

**Comment by JuantAldea** [comment]
Closing in favour of PR #3971.

**Pull request closed without merging**